File: src/dotnet/Orders.Domain/Models/Address.cs
using System.ComponentModel.DataAnnotations;

namespace Orders.Domain.Models;

public class Address : BaseEntity
{
    public string AddressLine { get; set; } = string.Empty;
    public int ZipCode { get; set; }
    public string City { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
}

File: src/client/src/models/address.ts
import faker from "@faker-js/faker";
import { Guid } from "guid-typescript";

export interface Address {
    id: string;
    addressLine: string;
    zipCode: string | number;
    city: string;
    country: string;
}

export class AddressFactory {
    static create(amount: number) {
        const addresses: Address[] = [];

        for (let i = 0; i < amount; i++) {
            const address: Address = {
                id: Guid.create().toString(),
                addressLine: faker.address.streetAddress(),
                zipCode: faker.address.zipCode(),
                city: faker.address.city(),
                country: faker.address.country()
            }

            addresses.push(address);
        }

        return addresses;
    }
}